vendor-extra/JatsContentBundle/tests/EventListener/BuildView/FrontArticleTitleTeaserListenerTest.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+declare(strict_types=1);
+
+namespace tests\Libero\JatsContentBundle\EventListener\BuildView;
+
+use Libero\JatsContentBundle\EventListener\BuildView\FrontArticleTitleTeaserListener;
+use Libero\ViewsBundle\Event\BuildViewEvent;
+use Libero\ViewsBundle\Views\TemplateView;
+use PHPUnit\Framework\TestCase;
+use tests\Libero\LiberoPageBundle\ViewConvertingTestCase;
+use tests\Libero\LiberoPageBundle\XmlTestCase;
+
+final class FrontArticleTitleTeaserListenerTest extends TestCase
+{
+    use ViewConvertingTestCase;
+    use XmlTestCase;
+
+    /**
+     * @test
+     * @dataProvider nodeProvider
+     */
+    public function it_does_nothing_if_it_is_not_a_jats_front_element(string $xml) : void
+    {
+        $listener = new FrontArticleTitleTeaserListener($this->createFailingConverter());
+
+        $element = $this->loadElement($xml);
+
+        $event = new BuildViewEvent($element, new TemplateView('@LiberoPatterns/teaser.html.twig'));
+        $listener->onBuildView($event);
+        $view = $event->getView();
+
+        $this->assertInstanceOf(TemplateView::class, $view);
+        $this->assertSame('@LiberoPatterns/teaser.html.twig', $view->getTemplate());
+        $this->assertEmpty($view->getArguments());
+        $this->assertEmpty($view->getContext());
+    }
+
+    public function nodeProvider() : iterable
+    {
+        yield 'different namespace' => ['<front xmlns="http://example.com">foo</front>'];
+        yield 'different element' => ['<foo xmlns="http://jats.nlm.nih.gov">foo</foo>'];
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_nothing_if_is_not_the_teaser_template() : void
+    {
+        $listener = new FrontArticleTitleTeaserListener($this->createFailingConverter());
+
+        $element = $this->loadElement(
+            <<<XML
+<?xml version="1.0" encoding="UTF-8"?>
+<front xmlns="http://jats.nlm.nih.gov">
+    <article-meta>
+        <title-group>
+            <article-title>foo</article-title>
+        </title-group>
+    </article-meta>
+</front>
+XML
+        );
+
+        $event = new BuildViewEvent($element, new TemplateView('template'));
+        $listener->onBuildView($event);
+        $view = $event->getView();
+
+        $this->assertInstanceOf(TemplateView::class, $view);
```

Comment:
I understand the negative cases but what would be a way of shortening the assertion phases of these tests? Extract a custom assertion method that contains this phase seems cheap and effective.

Reply:
`this->assertEquals(new TemplateView('template'), $view);` works.

Reply:
Exists in other tests too, will follow up with a refactor.